rest/client/methods.go

```diff
@@ -1040,18 +1041,23 @@ func (c *communicatorImpl) CreateVersionFromConfig(ctx context.Context, project,
 		method: http.MethodPut,
 		path:   "/versions",
 	}
+	var configAsYaml yaml.Node
+	err := yaml.Unmarshal(config, &configAsYaml)
+	if err != nil {
+		return nil, errors.Wrap(err, "reading provided yaml config")
+	}
 	body := struct {
-		ProjectID string          `json:"project_id"`
-		Message   string          `json:"message"`
-		Active    bool            `json:"activate"`
-		IsAdHoc   bool            `json:"is_adhoc"`
-		Config    json.RawMessage `json:"config"`
+		ProjectID string    `json:"project_id"`
+		Message   string    `json:"message"`
+		Active    bool      `json:"activate"`
+		IsAdHoc   bool      `json:"is_adhoc"`
+		Config    yaml.Node `json:"config"`
```

Comment:
I think it'd be preferable to simply send over the config byte array as part of the request and then unmarshaling it into yaml on the server side rather than converting it into a yaml client side and decoding it back into a byte array on the server side

Reply:
[Example](https://gist.github.com/hadjri/2a200f0686c6226d78d1550e098d2d77#file-distro-go-L125) of a route that uses a byte array directly